pkg/artifactory/resource_artifactory_ldap_setting.go

```diff
@@ -0,0 +1,286 @@
+	AutoCreateUser           bool           `xml:"autoCreateUser" yaml:"autoCreateUser"`
+	LdapPoisoningProtection  bool           `xml:"ldapPoisoningProtection" yaml:"ldapPoisoningProtection"`
+	AllowUserToAccessProfile bool           `xml:"allowUserToAccessProfile" yaml:"allowUserToAccessProfile"`
+	PagingSupportEnabled     bool           `xml:"pagingSupportEnabled" yaml:"pagingSupportEnabled"`
+	Search                   LdapSearchType `xml:"search" yaml:"search"`
+}
+
+type LdapSearchType struct {
+	SearchSubTree   bool   `yaml:"searchSubTree" xml:"searchSubTree"`
+	SearchFilter    string `yaml:"searchFilter" xml:"searchFilter"`
+	SearchBase      string `yaml:"searchBase" xml:"searchBase"`
+	ManagerDn       string `yaml:"managerDn" xml:"managerDn"`
+	ManagerPassword string `yaml:"managerPassword" xml:"managerPassword"`
+}
+
+type LdapSettings struct {
+	LdapSettingArr []LdapSetting `yaml:"ldapSetting" xml:"ldapSetting"`
+}
+
+type xmlLdapConfig struct {
+	XMLName      xml.Name     `xml:"config"`
+	LdapSettings LdapSettings `xml:"security>ldapSettings"`
+}
+
+func resourceArtifactoryLdapSetting() *schema.Resource {
+	return &schema.Resource{
+		UpdateContext: resourceLdapSettingsUpdate,
+		CreateContext: resourceLdapSettingsUpdate,
+		DeleteContext: resourceLdapSettingsDelete,
+		ReadContext:   resourceLdapSettingsRead,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"key": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "Ldap setting name",
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Flag to enable or disable the ldap setting",
+			},
+			"ldap_url": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "Location of the LDAP server in the following format: ldap://myserver:myport/dc=sampledomain,dc=com",
+			},
+			"user_dn_pattern": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "A DN pattern that can be used to log users directly in to LDAP. This pattern is used to create a DN string for 'direct' user authentication where the pattern is relative to the base DN in the LDAP URL. The pattern argument {0} is replaced with the username. This only works if anonymous binding is allowed and a direct user DN can be used, which is not the default case for Active Directory (use User DN search filter instead). Example: uid={0},ou=People",
+			},
+			"auto_create_user": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "When set, users are automatically created when using LDAP. Otherwise, users are transient and associated with auto-join groups defined in Artifactory.",
```

Comment:
Not specific to this attribute, but for optional attribute with default value in general. I feel we should include the default value in the description, e.g. "... Default to 'true'". Applies to documentation as well (at least until we auto generate).